server/manifest.go

```diff
@@ -46,7 +45,13 @@ func (m *Manifest) Remove() error {
 }
 
 func (m *Manifest) RemoveLayers() error {
-	for _, layer := range append(m.Layers, m.Config) {
+	layers := m.Layers
+
+	if m.Config.Digest != "" {
+		layers = append(layers, &m.Config)
+	}
+
+	for _, layer := range layers {
```

Comment:
```
for _, layer := range append(m.Layers, &m.Config) {
    if layer.Digest != "" {
    }
}
```

Reply:
This is probably a little cleaner and more robust.